modules/js-and-css/speculation-rules/helper.php

```diff
@@ -60,7 +60,15 @@ static function ( $exclude_path ) use ( $prefixer ) {
 		)
 	);
 
-	$prerender_rules = array(
+	$option = get_option( 'plsr_speculation_rules' );
+	if ( ! $option ) {
+		$option = plsr_get_setting_default();
+	}
```

Comment:
Alternative which would should the same result:

```suggestion
	$option = get_option( 'plsr_speculation_rules', plsr_get_setting_default() );
```

Reply:
Also, what about declaring the option with `add_option()`? The default value and the desired autoload behavior can then be specified explicitly.

Reply:
@westonruter Normally core will handle the default itself since this is a registered setting, i.e. passing the default isn't necessary. The fallback is only here for weird edge-cases (as far as I can tell only relevant for tests) where for some reason the option isn't registered. In other words, it's purely "very defensive coding".

> Also, what about declaring the option with `add_option()`? The default value and the desired autoload behavior can then be specified explicitly.

IMO not necessary. If a site uses the default, we don't have to store it in the DB. More importantly though, I prefer to avoid this as it would require some sort of "on activation" routine, which is generally something unreliable in WordPress plugins that doesn't scale well e.g. when using multisite. So IMO relying on a default is easier, also since we would want this option to be autoloaded anyway as it is small and used on pretty much every frontend page load.

Reply:
Is an activation routine needed? Isn't `add_option()` just run at `init`?

Reply:
Yeah, since `add_option()` writes to the database, it shouldn't be called in any regularly run action.

Reply:
When reviewing this PR, I had similar thoughts to what Weston proposed. Thanks for clarifying, Felix.